src/olympia/addons/models.py

```diff
@@ -1420,6 +1421,10 @@ def watch_status(old_attr=None, new_attr=None, instance=None,
     latest_version = instance.find_latest_version(
         channel=amo.RELEASE_CHANNEL_LISTED)
 
+    # Update the author's account profile visibility
+    if new_status != old_status:
+        [author.update_is_public() for author in instance.authors.all()]
```

Comment:
I wonder if this should happen in a celery task to not block for add-ons with many authors. Or just re-use `watch_addon_user` task?